System/Win32/Process.hsc

```diff
@@ -81,6 +81,15 @@ getCurrentProcessId = c_GetCurrentProcessId
 getCurrentProcess :: IO ProcessHandle
 getCurrentProcess = c_GetCurrentProcess
 
+foreign import WINDOWS_CCONV unsafe "windows.h TerminateProcess"
+    c_TerminateProcess :: ProcessHandle -> Int -> IO Int
+
+terminateProcessId :: ProcessId -> IO ()
+terminateProcessId p = bracket
+    (openProcess pROCESS_TERMINATE False p)
+    closeHandle
+    (\h -> void $ c_TerminateProcess h 1)
```

Comment:
don't swallow the exception, change `void` to `failIfFalse_ "c_TerminateProcess"`